### PR TITLE
ENH: -r/--remote for list-handles (Closes #269)

### LINKED ISSUE #269: How to list available (remote) handles?
may be there is a command already but couldn't find.

E.g. after I registered a remote collection, how could I discover what it provides?  I can envision having e.g. `list-handles -r [collection]` which would list all handles available from remote collections  (or just from `[collection]`) or a new command `list-collection [collection]` which does that

### PR DESCRIPTION
**Pull request by yarikoptic**
I wish I could make a reasonable unittest but I think we better address #234, etc for that

### CODE REVIEW

File: datalad/interface/list_handles.py [hunk 1]
```diff
@@ -12,25 +12,41 @@
...
-            print(handle)
+        
+        if remote:
+            for remote_branch in local_master.git_get_remote_branches():
+                if not remote_branch.endswith('/master'): # for now only those
+                    continue
+                for h in CollectionRepoBackend(local_master, branch=remote_branch).get_handles():
+                    remote_name = '/'.join(remote_branch.split('/')[:-1])
+                    print("%s/%s" % (remote_name, h))
+        else:
```
Comment (bpoldrack) [inline datalad/interface/list_handles.py]: Is it really intended to be "else"? I'd expect a "-r" to ALSO list remote handles, not JUST remote handles.

Comment (yarikoptic) [inline datalad/interface/list_handles.py]: it is not '--all' but '--remote' so I felt this way.  Modes of operation are either to list installed handles or available (--remote) not all the handles at once (I don't see when I would need both)

Comment (bpoldrack) [inline datalad/interface/list_handles.py]: Well, so let's have both options. I just FEEL it's incomplete, if you can
list local ones, remote ones, but not both of them. ;)

2015-10-17 17:00 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

Comment (yarikoptic) [inline datalad/interface/list_handles.py]: I would not resist if someone implements it but myself I don't see using it yet... ;-)

**Pull request merged**